code/modules/sealant_gun/sealant_injector.dm

```diff
@@ -0,0 +1,94 @@
+/obj/item/chems/chem_disp_cartridge/foaming_agent
+	spawn_reagent = /decl/material/liquid/foaming_agent
+/obj/item/chems/chem_disp_cartridge/polyacid
+	spawn_reagent = /decl/material/liquid/acid/polyacid
+
+/obj/structure/sealant_injector
+	name = "sealant tank injector"
+	icon = 'icons/obj/structures/sealant_props.dmi'
+	icon_state = "injector"
+
+	var/list/cartridges
+	var/obj/item/sealant_tank/loaded_tank
```

Comment:
Clear this stuff on `Destroy`. If you want to be good, listen for their Destroy events as well in case of explosions, etc.